Create a program using a for loop that prompts the user to enter five numbers. After all numbers are entered, the program should display the average of the numbers.

In [2]:
fruits = ["apple", "banana", "cherry"]

convert this list into a tuple

In [3]:
fruits_tuple = tuple(fruits)

add orange to the list of fruits

In [4]:
fruits.append("orange")

show the difference in the content of the list and tuple after the addition

In [5]:
# Create the list and tuple
fruits = ["apple", "banana", "cherry"]
fruits_tuple = tuple(fruits)

# Display the content before adding "orange"
print("List (fruits) before:", fruits)
print("Tuple (fruits_tuple) before:", fruits_tuple)

# Add "orange" to the list
fruits.append("orange")

# Display the content after adding "orange"
print("List (fruits) after:", fruits)
print("Tuple (fruits_tuple) after:", fruits_tuple)

List (fruits) before: ['apple', 'banana', 'cherry']
Tuple (fruits_tuple) before: ('apple', 'banana', 'cherry')
List (fruits) after: ['apple', 'banana', 'cherry', 'orange']
Tuple (fruits_tuple) after: ('apple', 'banana', 'cherry')


create a dictionary named person with keys "first name", "last name", and "age".

In [6]:
person = {
    "first name": "John",
    "last name": "Doe",
    "age": 30
}

add a new key-value pair, "city" with a value of your choice.

In [7]:
person["city"] = "New York"

display the keys in the dictionary using a python command

In [8]:
keys = person.keys()
print(keys)

dict_keys(['first name', 'last name', 'age', 'city'])


Create a set named colors with the values "red", "blue", "green".

In [9]:
colors = {"red", "blue", "green"}

Add "yellow" to the set and remove "blue".

In [10]:
colors.add("yellow")
colors.remove("blue")

Check if "pink" is in the set and display the result.

In [11]:
if "pink" in colors:
    print("Yes, 'pink' is in the set.")
else:
    print("No, 'pink' is not in the set.")

No, 'pink' is not in the set.


Write a Python program that takes an input from the user and checks if the number is positive, negative, or zero.

In [12]:
# Get user input as a float
try:
    number = float(input("Enter a number: "))

    # Check if the number is positive, negative, or zero
    if number > 0:
        print("The number is positive.")
    elif number < 0:
        print("The number is negative.")
    else:
        print("The number is zero.")
except ValueError:
    print("Invalid input. Please enter a valid number.")

Enter a number: 10
The number is positive.


Given a list of numbers: [23, 45, 8, 12, 78, 56, 89], write a Python program that prints out all even numbers from the list using a conditional statement.

In [13]:
# Define the list of numbers
numbers = [23, 45, 8, 12, 78, 56, 89]

# Iterate through the list and print even numbers
print("Even numbers in the list:")
for num in numbers:
    if num % 2 == 0:
        print(num)

Even numbers in the list:
8
12
78
56


Write a program that asks the user to enter their age. Based on the user's input, determine if they are eligible to vote (consider the voting age as 18).

In [14]:
# Get user input for age
try:
    age = int(input("Enter your age: "))

    # Check if the user is eligible to vote
    if age >= 18:
        print("You are eligible to vote.")
    else:
        print("You are not eligible to vote.")
except ValueError:
    print("Invalid input. Please enter a valid age as a number.")

Enter your age: 20
You are eligible to vote.


Write a program that prompts the user to enter numbers continuously (using a while loop) and stops taking inputs when the user enters 'q'. After that, display the sum of all the numbers entered.

In [15]:
# Initialize variables
total = 0

# Continuously prompt the user for input
while True:
    user_input = input("Enter a number (or 'q' to quit): ")

    # Check if the user wants to quit
    if user_input.lower() == 'q':
        break

    # Attempt to convert the input to a number
    try:
        number = float(user_input)
        total += number
    except ValueError:
        print("Invalid input. Please enter a number or 'q' to quit.")

# Display the sum of numbers entered
print("Sum of the numbers:", total)

Enter a number (or 'q' to quit): 20
Enter a number (or 'q' to quit): q
Sum of the numbers: 20.0


Create a Python program using a while loop that prints numbers from 10 down to 1 and then prints "Liftoff!".

In [16]:
# Initialize the countdown starting point
count = 10

# Use a while loop to count down
while count >= 1:
    print(count)
    count -= 1

# Print "Liftoff!" after the countdown
print("Liftoff!")

10
9
8
7
6
5
4
3
2
1
Liftoff!


Create a program that asks the user for a number and then prints out the multiplication table for that number using a while loop.

In [17]:
# Get user input for the number
try:
    number = int(input("Enter a number: "))

    # Initialize a counter
    i = 1

    # Use a while loop to print the multiplication table
    while i <= 10:
        result = number * i
        print(f"{number} x {i} = {result}")
        i += 1
except ValueError:
    print("Invalid input. Please enter a valid number.")

Enter a number: 20
20 x 1 = 20
20 x 2 = 40
20 x 3 = 60
20 x 4 = 80
20 x 5 = 100
20 x 6 = 120
20 x 7 = 140
20 x 8 = 160
20 x 9 = 180
20 x 10 = 200


Given a list: ["apple", "banana", "cherry", "date"], use a for loop to print each fruit in the list.

In [18]:
fruits = ["apple", "banana", "cherry", "date"]

# Use a for loop to print each fruit
for fruit in fruits:
    print(fruit)

apple
banana
cherry
date


Using a for loop, calculate the factorial of a given number (n!). For example, if the given number is 5, the output should be 5 x 4 x 3 x 2 x 1 = 120.

In [19]:
# Get user input for the number
try:
    n = int(input("Enter a number: "))

    # Initialize a variable to store the factorial
    factorial = 1

    # Use a for loop to calculate the factorial
    for i in range(1, n + 1):
        factorial *= i

    # Print the result
    print(f"{n}! = {factorial}")
except ValueError:
    print("Invalid input. Please enter a valid number.")


Enter a number: 20
20! = 2432902008176640000


Create a program using a for loop that prompts the user to enter five numbers. After all numbers are entered, the program should display the average of the numbers.

In [20]:
# Initialize a variable to store the sum of the numbers
total = 0

# Use a for loop to prompt the user for five numbers
for i in range(5):
    try:
        number = float(input(f"Enter number {i + 1}: "))
        total += number
    except ValueError:
        print("Invalid input. Please enter a valid number.")

# Calculate the average
average = total / 5

# Display the average
print(f"The average of the five numbers is: {average}")

Enter number 1: 20
Enter number 2: 5
Enter number 3: 4
Enter number 4: 3
Enter number 5: 2
The average of the five numbers is: 6.8
